packages/core/src/useDescriptors.tsx

```diff
@@ -116,43 +121,47 @@ export default function useDescriptors<
     (acc, route) => {
       const screen = screens[route.name];
       const navigation = navigations[route.key];
+      const routeOptions: ScreenOptions & SharedScreenNavigationOptions = {
+        // The default `screenOptions` passed to the navigator
+        ...(typeof screenOptions === 'object' || screenOptions == null
+          ? screenOptions
+          : screenOptions({
+              // @ts-ignore
+              route,
+              navigation,
+            })),
+        // The `options` prop passed to `Screen` elements
+        ...(typeof screen.options === 'object' || screen.options == null
+          ? screen.options
+          : screen.options({
+              // @ts-ignore
+              route,
+              // @ts-ignore
+              navigation,
+            })),
+        // The options set via `navigation.setOptions`
+        ...options[route.key],
+      };
 
       acc[route.key] = {
         navigation,
         render() {
           return (
             <NavigationBuilderContext.Provider key={route.key} value={context}>
-              <SceneView
-                navigation={navigation}
-                route={route}
-                screen={screen}
-                getState={getState}
-                setState={setState}
-              />
+              <NavigationContext.Provider value={navigation}>
```

Comment:
The `NavigationContext` is already there in `SceneView` so changes to this file shouldn't be needed.

Reply:
I moved it from scene view here to have it accessible in ScreeView component